docs/src/main/asciidoc/websockets-next-reference.adoc

```diff
@@ -0,0 +1,546 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= WebSockets Next Extension Reference Guide
+:extension-status: preview
+include::_attributes.adoc[]
+:numbered:
+:sectnums:
+:categories: web
+:topics: web,websockets
+:extensions: io.quarkus:quarkus-websockets-next
+
+== The WebSocket protocol
+
+The _WebSocket_ protocol, documented in the https://datatracker.ietf.org/doc/html/rfc6455[RFC6455], establishes a standardized method for creating a bidirectional communication channel between a client and a server through a single TCP connection.
+Unlike HTTP, WebSocket operates as a distinct TCP protocol but is designed to function seamlessly alongside HTTP.
+For example, it reuses the same ports and is compatible with the same security mechanisms.
+
+The interaction using WebSocket initiates with an HTTP request employing the 'Upgrade' header to transition to the WebSocket protocol.
+Instead of a `200 OK` response, the server replies with a `101 Switching Protocols` response to upgrade the HTTP connection to a WebSocket connection.
+Following this successful handshake, the TCP socket utilized in the initial HTTP upgrade request remains open, allowing both client and server to exchange messages in both direction continually.
+
+== HTTP and WebSocket architecture styles
+
+Despite WebSocket's compatibility with HTTP and its initiation through an HTTP request, it's crucial to recognize that the two protocols lead to distinctly different architectures and programming models.
+
+With HTTP/REST, applications are structured around resources/endpoints that handle various HTTP methods and paths.
+Client interaction occurs through emitting HTTP requests with appropriate methods and paths, following a request-response pattern.
+The server routes incoming requests to corresponding handlers based on path, method, and headers and then replies with a well-defined response.
+
+Conversely, WebSocket typically involves a single endpoint for the initial HTTP connection, after which all messages utilize the same TCP connection.
+It introduces an entirely different interaction model: asynchronous and message-driven.
+
+WebSocket is a low-level transport protocol, in contrast to HTTP.
+Message formats, routing, or processing require prior agreement between the client and server regarding message semantics.
+
+For WebSocket clients and servers, the `Sec-WebSocket-Protocol` header in the HTTP handshake request allows negotiation of a higher-level messaging protocol. In its absence, the server and client must establish their own conventions.
```

Comment:
This reminds me that we might add something convenient here as well. Currently, the only thing you can do is obtain the header value with `connection.handshakeRequest().header("Sec-WebSocket-Protocol")`.

Reply:
Can you open an issue so we do not forget?